include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -1089,9 +1086,9 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin,
     for (ptrdiff_t i = 0u; i < assetCount; ++i)
     {
         const asset::ICPUImage* cpuimg = _begin[i];
-        asset::IImage::SCreationParams params = cpuimg->getCreationParameters();
-        params.initialLayout = asset::EIL_UNDEFINED;
-        params.sharingMode = _params.sharingMode;
+        IGPUImage::SCreationParams params = {};
+        static_cast<asset::IImage::SCreationParams>(params) = cpuimg->getCreationParameters();
```

Comment:
Wait what? this looks ugly, Can I add an `=` operator for IGPUImage::SCreationParams to IImage::SCreationParams?

Reply:
yes you can, I also fucked up because I didn't use `&` in the static_cast